common/src/main/java/com/wynntils/core/config/ConfigManager.java

```diff
@@ -37,61 +37,51 @@ public final class ConfigManager extends Manager {
+    public ConfigManager(
+            ConfigUpfixerManager configUpfixerManager,
+            JsonManager jsonManager,
+            FeatureManager feature,
+            OverlayManager overlay) {
+        super(List.of(configUpfixerManager, jsonManager, feature, overlay));
 
         userConfig = new File(CONFIGS, McUtils.mc().getUser().getUuid() + FILE_SUFFIX);
+    }
 
+    public void init() {
         // First, we load the config file
         configObject = Managers.Json.loadPreciousJson(userConfig);
 
         // Now, we have to apply upfixers, before any config loading happens
-        if (configUpfixerManager.runUpfixers(configObject)) {
+        if (Managers.ConfigUpfixer.runUpfixers(configObject)) {
             Managers.Json.savePreciousJson(userConfig, configObject);
         }
-    }
-
-    public void registerFeature(Feature feature) {
-        registerOverlayGroups(feature);
-
-        for (Overlay overlay : feature.getOverlays()) {
-            registerConfigOptions(overlay);
-        }
 
-        registerConfigOptions(feature);
-    }
+        // Register all features and overlays
+        Managers.Feature.getFeatures().forEach(this::registerFeature);
 
-    private void registerOverlayGroups(Feature feature) {
-        List<OverlayGroupHolder> holders = Stream.of(feature.getClass().getDeclaredFields())
-                .filter(f -> f.isAnnotationPresent(OverlayGroup.class))
-                .map(field -> new OverlayGroupHolder(
-                        field, feature, field.getAnnotation(OverlayGroup.class).instances()))
-                .toList();
+        // Finish off the config init process
 
-        if (holders.isEmpty()) return;
+        // Load configs for all features
+        Managers.Config.reloadConfiguration();
 
-        OVERLAY_GROUP_FIELDS.addAll(holders);
-        feature.addOverlayGroups(holders);
+        // Save config file after loading all configurables' options
+        Managers.Config.saveConfig();
 
-        for (OverlayGroupHolder holder : holders) {
-            holder.initGroup(
-                    IntStream.rangeClosed(1, holder.getDefaultCount()).boxed().toList());
-
-            List<ConfigHolder> overlayHolders = holder.getOverlays().stream()
-                    .map(this::getConfigOptions)
-                    .flatMap(List::stream)
-                    .toList();
-
-            holder.getOverlays().forEach(overlay -> overlay.addConfigOptions(this.getConfigOptions(overlay)));
+        // Create default config file containing all configurables' options
+        Managers.Config.saveDefaultConfig();
+    }
 
-            OVERLAY_GROUP_CONFIG_HOLDERS.addAll(overlayHolders);
+    public void registerFeature(Feature feature) {
+        for (Overlay overlay : Managers.Overlay.getFeatureOverlays(feature).stream()
+                .filter(overlay -> !overlay.isDynamic())
+                .toList()) {
+            registerConfigOptions(overlay);
         }
+
```

Comment:
This is perhaps a bit irrational, but I think it would make more sense to start with the registration of the feature itself, and then do the overlays, perhaps with a comment `// also register the overlays of the feature` or so. (I know this is the same order as it was before, but I was always like flummoxed by that function as well before).